docs/essentials/configuring-pre-flight-checks.md

```diff
@@ -0,0 +1,130 @@
+
+5. Click **Save changes** button.
+
+To remove pre-flight checks for the project, just click **Delete pre-flight checks**
+button, which should be visible if you have pre-flight checks configured.
+
+## Examples of pre-flight checks configuration
+
+### Limiting using secrets in your organization
+
+In the initialization job, the following information is available as environment variables:
+
+- [the project name](/ci-cd-environment/environment-variables/#semaphore_project_name) for which the pipeline started
+- [repository branch](/ci-cd-environment/environment-variables/#semaphore_git_branch) for which the pipeline started
+- YAML file describing the running pipeline (`SEMAPHORE_YAML_FILE_PATH`)
+
+Based on that, you can define commands in pre-flight checks ensuring that
+a particular pipeline can use certain secrets. You can use this [snippet](https://gist.githubusercontent.com/shiroyasha/49b57770371f4684a8a12bcec0047d7a/raw/b35e51953069dc1b21c62a41b8717aa5fb4bcf88/check-secret)
+as a reference on how to achieve that result. In practice, using it for restricting 
+`deployment-secret` secret for pipeline defined in `.semaphore/deployment.yml`
+in the `example-project` on the `master` branch comes down to the following commands:
+
+```bash
+curl https://gist.githubusercontent.com/shiroyasha/49b57770371f4684a8a12bcec0047d7a/raw/b35e51953069dc1b21c62a41b8717aa5fb4bcf88/check-secret -o check-secret
+
+bash check-secret "deployment-secret" "example-project" ".semaphore/deployment.yml" "master"
+```
+
+You can also combine it with [secrets](/essentials/using-secrets/) or cloned GitOps repository,
+as well as other [Semaphore Environment Variables](/ci-cd-environment/environment-variables)
+to fine-tune the commands to your particular use case. 
+
+### Limiting triggering promotions in your organization
+
+In the initialization job, the following information is available as environment variables:
+
+- [GitHub repository](/ci-cd-environment/environment-variables/#semaphore_git_repo_slug) for which the pipeline started
+- [repository branch](/ci-cd-environment/environment-variables/#semaphore_git_branch) for which the pipeline started
+- [boolean flag](/ci-cd-environment/environment-variables/#semaphore_pipeline_promotion) describing if the pipeline is an initial one or a promotion
+- [GitHub username](/ci-cd-environment/environment-variables/#semaphore_pipeline_promoted_by) of a person triggering the promotion
+
+Based on that, you can define commands in pre-flight checks ensuring that a person who 
+started a promotion is entitled to do so. As an example, we'll demonstrate how to restrict
+the possibility of promoting to a selected group of users:
+
+```bash
+printf '%s\n' 'user-1' 'user-2' 'user-3' > allowed_users.txt
+user_is_allowed () { grep -Fxq $SEMAPHORE_PIPELINE_PROMOTED_BY allowed_users.txt; }
+is_promotion () { [ $SEMAPHORE_PIPELINE_PROMOTION == "true" ]; }
+if is_promotion && user_is_allowed; then echo "Promotion allowed."; else false; fi
```

Comment:
On a second look, this will fail for all pipelines that are **not** promotions.
We need to change it to:
```
if is_promotion; then if user_is_allowed; then echo "Promotion allowed."; else false; fi; else echo "Initial pipelines are allowed." fi
```